cost/aws/reserved_instances/utilization/utilization_ris.pt

```diff
@@ -65,47 +200,85 @@ datasource "ds_reservations" do
+  result "result"
   code <<-EOS
-  // This is the list of filtered volumes.
-  var modified_reservations = [];
-  for (var i = 0; i < reservations.length; i++) {
-    modified_reservations.push(
-      {
-        utilization_percentage: Math.round(parseFloat(reservations[i]["utilization_percentage"])),
-        end_datetime: reservations[i]["end_datetime"],
-        start_datetime: reservations[i]["start_datetime"],
-        account_name: reservations[i]["account_name"],
-        account_id: reservations[i]["account_id"],
-        region: reservations[i]["region"],
-        instance_type: reservations[i]["instance_type"],
-        instance_count: reservations[i]["instance_count"],
-        scope: reservations[i]["scope"],
-        type: reservations[i]["type"],
-      }
-    )
-  }
-  modified_reservations = _.sortBy(modified_reservations, 'region');
-EOS
+
+  //Filter reservations for billing centers that contain a matching account ID
+  var bc_account_ids = _.uniq(_.pluck(ds_accounts_by_billing_center, "vendor_account_id"))
+  var filtered_reservations = _.filter(ds_reservations, function(ri) {
+    return _.contains(bc_account_ids, ri.account_id);
+  })
+
+  result = filtered_reservations
+  EOS
+end
+
+#ENRICH FILTERED RESERVATION DATA WITH BILLING CENTER DATA
+datasource "ds_enriched_reservation_data" do
+	run_script $js_enrich_reservation_data_with_bc_data, $ds_filtered_reservations, $ds_accounts_by_billing_center, $ds_filtered_billing_centers
+end
+
+script "js_enrich_reservation_data_with_bc_data", type: "javascript" do
+	parameters "ds_filtered_reservations", "ds_accounts_by_billing_center", "ds_filtered_billing_centers"
+	result "result"
+	code <<-EOS
+		_.each(ds_filtered_reservations, function(ri) {
+
+			//Get associated billing centers for reservation
+			var associated_billing_centers = _.filter(ds_accounts_by_billing_center, function(bc) {
+        return bc.vendor_account_id == ri.account_id
+			})
+			
+			//Get list of associated billing center names and convert the list to a string
+			var list_of_associated_bcs_string = ""
+			_.each(associated_billing_centers, function(bc) {
+        var billing_center_name = ""
+        var billing_center_details = _.find(ds_filtered_billing_centers, function(fbc) { return bc.billing_center_id == fbc.id })
+        if (billing_center_details != undefined) {
+          billing_center_name = billing_center_details.name
+        }
+
+				if (list_of_associated_bcs_string == "") {
+					list_of_associated_bcs_string = billing_center_name
+				} else {
+					list_of_associated_bcs_string += ", " + billing_center_name
+				}	
+			})
+
+			//Add list of associated billing center names string to reservation object for policy incident
+			ri["billing_centers"] = list_of_associated_bcs_string
+		})
+
+		result = ds_filtered_reservations
+	EOS
 end
 
-escalation "report_reserved_instances_utilization" do
+###############################################################################
```

Comment:
Escalation section should be below Policy section.

Reply:
Done